Mention need to add dependencies in rosidl_generate_interfaces

NikolasE opened the pull request:
Added hint to list dependencies for used packages in rosidl_generate_interfaces. Otherwise the user will only get this message

"rosidl_generator_py.import_type_support_impl.UnsupportedTypeSupport: Could not import 'rosidl_typesupport_c' for package"

that's not really helpful

NikolasE committed:
Update Custom-ROS2-Interfaces.rst

Added hint to list dependencies for used packages

fujitatomoya reviewed (commented):
I think this chapter is all about `How to add custom interfaces`, so i am not really inclined to have this change in general. if we do this, i would add `DEPENDENCIES builtin_interfaces` instead.

NikolasE commented:
Most custom interfaces use existing messages from other packages, so I think the hint is definitely important. And geometry_msgs was just an example for a package that everyone knows so I thought it would be a good example.

fujitatomoya commented:
thanks! sounds good to me. probably `how to use existing message or service type` would be good for user manual.

audrow reviewed (commented):
Thanks for the suggestion, @NikolasE. I agree that this is a challenging error message to understand. I'm more in favor of the following two things:
- improving the error message in `rosidl_generate_interfaces` (maybe it's worth opening an issue there, if there's not already one)
- adding a note after the code example you've modified and saying something like, "Make sure to remember to add dependencies that your interface depends on." You could also have an example with the `DEPENDENCIES` line in it.

NikolasE committed:
Update Custom-ROS2-Interfaces.rst

added example for a message that uses a non-basic data type

NikolasE commented:
@audrow I added a message that references geometry_msgs.

clalancette commented on source/Tutorials/Custom-ROS2-Interfaces.rst:
```diff
@@ -71,6 +71,14 @@ In the ``tutorial_interfaces/msg`` directory you just created, make a new file c
+.. code-block:: console
+
+    geometry_msgs/Point center
+    float64  radius
```
```suggestion
    float64 radius
```

clalancette reviewed (changes requested):
I've left some changes inline.

That said, I actually don't think that this information belongs in this tutorial at all.  This is a beginner tutorial, and the point of the beginner tutorials is to show simple ways to do things in ROS 2.  It is not meant to show every way to do things, and I think this strays into that territory.

My suggestion is to create either a separate tutorial or a separate How-To Guide that explains how to add dependencies to messages.  And then link to that from the end of this tutorial in a "more information" section.

NikolasE committed:
Update Custom-ROS2-Interfaces.rst

fixed typos

fujitatomoya commented:
@clalancette IMO, tutorial is about `How to add custom message`, and it is likely that custom message depends on other message packages. so i think it is okay to have this change in this section.

@NikolasE would you change the target branch to `rolling`?

clalancette committed:
Lots of cleanups.

All of this additional information is needed for the tutorial
to actually work.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette reviewed (approved):
It seems like people agree that we need this kind of information here.

However, I had to go through and make a number of edits to make sure that the tutorial still actually worked after this change.  This now seems to work for me.
